unit_tests/tests/test_call.rs

```diff
@@ -0,0 +1,299 @@
+ */
+#[rstest]
+#[tokio::test]
+async fn fail_invalid_contract_call_data(clients: HashMap<String, JsonRpcClient<HttpTransport>>) {
+    let deoxys = &clients[DEOXYS];
+
+    let response_deoxys = deoxys.call(
+        FunctionCall {
+            contract_address: FieldElement::from_hex_be(STARKGATE_ETH_BRIDGE_ADDR).unwrap(),
+            entry_point_selector: get_selector_from_name("balanceOf").unwrap(),
+            calldata: vec![
+                FieldElement::ZERO
+            ]
+        },
+        BlockId::Tag(BlockTag::Latest)
+    ).await.expect("Error waiting for response from Deoxys node");
+
+    assert_eq!(response_deoxys, vec![FieldElement::ZERO, FieldElement::ZERO]);
+}
+
+/**
+ * Unit test for `starknet_call`
+ * 
+ * purpose: function request `name` to StarkGate ETH bridge contract
+ * fail case: too many arguments in call data
+ */
+#[rstest]
+#[tokio::test]
+async fn fail_too_many_call_data(clients: HashMap<String, JsonRpcClient<HttpTransport>>) {
+    let deoxys = &clients[DEOXYS];
+
+    let response_deoxys = deoxys.call(
+        FunctionCall { 
+            contract_address: FieldElement::from_hex_be(STARKGATE_ETH_BRIDGE_ADDR).unwrap(),
+            entry_point_selector: get_selector_from_name("name").unwrap(), 
+            calldata: vec![
+                FieldElement::ZERO
+            ]
+        },
+        BlockId::Tag(BlockTag::Latest)
+    ).await.err();
+
+    assert_matches!(
+        response_deoxys,
+        Some(ProviderError::StarknetError(StarknetErrorWithMessage {
+            message: _,
+            code: MaybeUnknownErrorCode::Known(StarknetError::ContractError)
+        }))
+    );
+}
+
+/**
+ * Unit test for `starknet_call`
+ * 
+ * purpose: function request `name` to StarkGate ETH bridge contract
+ * success case: should return 'Ether'
+ */
+#[rstest]
+#[tokio::test]
+async fn work_correct_call(clients: HashMap<String, JsonRpcClient<HttpTransport>>) {
+    let deoxys = &clients[DEOXYS];
+    let pathfinder = &clients[PATHFINDER];
+
+    let response_deoxys = deoxys.call(
+        FunctionCall { 
+            contract_address: FieldElement::from_hex_be(STARKGATE_ETH_BRIDGE_ADDR).unwrap(),
+            entry_point_selector: get_selector_from_name("name").unwrap(), 
+            calldata: vec![]
+        },
+        BlockId::Tag(BlockTag::Latest)
+    ).await.expect("Error waiting for response from Deoxys node");
+
+    let response_pathfinder = pathfinder.call(
+        FunctionCall { 
+            contract_address: FieldElement::from_hex_be(STARKGATE_ETH_BRIDGE_ADDR).unwrap(), 
+            entry_point_selector: get_selector_from_name("name").unwrap(), 
+            calldata: vec![]
+        },
+        BlockId::Tag(BlockTag::Latest)
+    ).await.expect("Error waiting for response from Pathfinder node");
+
+    let response_expected = short_string!("Ether");
+
+    assert_eq!(response_deoxys, vec![response_expected]);
```

Comment:
Could you tell me why you're testing this?

Reply:
It is possible for `response_deoxys == response_pathfinder` to evaluate to `true` in case of a call with invalid call_data. This means that there need to be two checks:

1. correct data has been received for Deoxys
2. data received by Deoxys matches that of Pathfinder